cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -79,7 +79,92 @@
 import javax.annotation.Nullable;
 
 /**
- * Dataset that manages Metadata using an {@link IndexedTable}.
+ * Dataset that manages Metadata using an {@link IndexedTable}. Supports writing tags and properties. Tags are single
+ * elements, whereas properties are key-values.
+ *
+ * Tags and properties both get translated into a {@link MetadataEntry}. A {@link MetadataEntry} corresponds to all
```

Comment:
Minor correction: MetadataEntry does not correspond to all the metadata which is set on same MetadataEntity it is just one metadata entry. Metadata represent all the metadata of a cdap resource.

Reply:
done